spec/mock-request.js

```diff
@@ -57,23 +78,27 @@ HttpBackend.prototype = {
                 flushed += 1;
                 if (numToFlush && flushed === numToFlush) {
                     console.log("  [%s] Flushed assigned amount: %s", path, numToFlush);
+                    // Restore Clock
+                    if (self.useMockClock) {
+                        callbacks.setNow();
+                    }
                     defer.resolve();
                 } else {
-                    setTimeout(tryFlush, 0);
+                    setImmediate(tryFlush);
                 }
-            } else if (flushed === 0 && !triedWaiting) {
-                // we may not have made the request yet, wait a generous amount of
-                // time before giving up.
-                setTimeout(tryFlush, 5);
-                triedWaiting = true;
+            } else if (flushed === 0 && waitLoops < 50) {
+                waitLoops += 1
```

Comment:
I'd like very much to improve this rather than committing it to the code-base (haven't done a lot of testing in Node), are you aware of a better strategy or a reference code-base to take a peek at?